### ISSUE TITLE
Checking more in CI

### ISSUE THREAD

**Issue by CosmicHorrorDev** [trimental/inlyne#142]
Wanted to run potential changes by @trimental before starting on work. I figured that it could be good to add the following to CI

- Clippy - We only have on lint currently, and it's been that way for a while. It's just a constructor with a lot of unique types, so seems fine to just `#[allow()]` that one instance
- Formatting - It looks like both of us probably already use formatting on save, so this mostly just makes sure that one-off contributions also handle formatting
- Publishing code coverage on something like [codecov.io](https://about.codecov.io) - I normally don't enable the integration that comments on every PR since it's annoying, but it's still nice to have the interface for poking around to see what is covered and how coverage has changed over time

**Comment by CosmicHorrorDev** [comment]
The only task left here is codecov.io. I believe that configuring that requires @trimental to setup things on their end, so I'll close this till he's active again
